awswrangler/s3/_select.py

```diff
@@ -0,0 +1,216 @@
+"""Amazon S3 Select Module (PRIVATE)."""
+
+import concurrent.futures
+import itertools
+import json
+import logging
+import pprint
+from typing import Any, Dict, List, Optional, Tuple, Union
+
+import boto3
+import pandas as pd
+
+from awswrangler import _utils, exceptions
+from awswrangler.s3._describe import size_objects
+
+_logger: logging.Logger = logging.getLogger(__name__)
+
+_RANGE_CHUNK_SIZE: int = int(1024 * 1024)
```

Comment:
Reading 1 MB per scan range

Reply:
Not that I'd like to encourage encouraging this behaviour but users can always monkey patch this really required, and we're not exposing these in other places across codebase. 
I say we keep it as is and add ability to override when user start asking for it.